### PR TITLE
Handle case in factory reset on Linux where pkill has exit code 1

### LINKED ISSUE #3619: On Linux, factory resets fail with message `Error: failed to kill qemu: exit status 1`
`pkill` is executed in order to kill QEMU when doing a factory reset. If the QEMU process is not found, `pkill` has an exit code of 1, which `rdctl` interprets as an error.

### PR DESCRIPTION
**Pull request by adamkpickering**
Fixes #3619.

### CODE REVIEW

**Review by mook-as — CHANGES_REQUESTED** [review]
Please also fix `pkillDarwin` and `pkillLinux`. Sorry about introducing the bug.

File: src/go/rdctl/pkg/shutdown/shutdown.go [hunk 1]
```diff
@@ -117,7 +117,13 @@ func pkillQemu() error {
 	cmd.Stderr = os.Stderr
 	err := cmd.Run()
 	if err != nil {
-		return fmt.Errorf("failed to kill qemu: %w", err)
+		if exitError, ok := err.(*exec.ExitError); ok {
+			if exitCode := exitError.ExitCode(); exitCode != 0 || exitCode != 1 {
```
Comment (ericpromislow) [inline src/go/rdctl/pkg/shutdown/shutdown.go:121]: `exitCode != 0 || exitCode != 1` is always true.
Using deMorgan: ~ => `!(exitCode == 0 && exitCode ==1)` => `!<always false>` => `true`

Given that, you'll need a comment here to explain what you're looking for

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Needs a comment, probably mentioning that the process we're trying to kill might have already died after we verified it was still running.

File: src/go/rdctl/pkg/shutdown/shutdown.go (hunk 1 above)
Comment (adamkpickering) [reply src/go/rdctl/pkg/shutdown/shutdown.go:121]: Wow, totally missed that. Nice catch. I meant `exitCode != 0 && exitCode != 1`.

**Review by ericpromislow — DISMISSED** [review]
Looks good now

**Review by mook-as — DISMISSED** [review]

**Review by mook-as — APPROVED** [review]
Given FHS + usr-merge, `/usr/bin/pkill` should be safe on Linux too, but at this point we should probably not do that until later :)

**Pull request merged**